sections/builder.js

```diff
@@ -283,15 +283,21 @@ export default [
   {
     type: "method",
     method: "with",
-    example: ".with(alias, function|raw)",
-    description: "Add a \"with\" clause to the query. \"With\" clauses are supported by PostgreSQL, Oracle, SQLite3 and MSSQL.",
+    example: ".with(alias, [*columns], callback|builder|raw)",
```

Comment:
btw, wouldn't it be more correct that third parameter is optional, and second one could be either `columns`, or whatever third parameter could be as well?

Reply:
I think it’s equally correct, but I worry it obscures the intended behavior due to the complexity of the type signature:

    with(alias, columns | (qb|callback|raw), (qb|callback|raw) | undefined)

The simplest way to document this IMO would be as two functions with different arities - which is how the TypeScript types are written, in fact. I don’t know that the documentation generator supports that notion of overloaded functions directly, though?

Reply:
I doubt it, so your original format might be better